### ISSUE TITLE
Support for Develco Smoke Alarm SMSZB-120

### ISSUE THREAD

**Issue by BlizzWave** [Koenkk/zigbee2mqtt#3012]
The original issue has gone stale, but is still not resolved. 
#1809 

Help with getting this smoke alarm to work with zigbee2mqtt.

Link to log

[Log](https://pastebin.com/rsHx5uzA)

**Comment by Koenkk** [comment]
- First make sure you are running zigbee2mqtt 1.11.0 (contains interview improvements)
- Let's then focus on getting the smoke detection to work. Could you provide the logging when triggering the smoke alarm?

**Comment by BlizzWave** [comment]
I have made some progress here. 
Smoke detection works and battery related stuff works. 

I still don't get a successful interview though.
And i don't know how to trigger the siren with command. 

```
{
"smoke":false,"
battery_low":false,
"linkquality":97,
"battery":"100.00",
"voltage":3500,
"battery_alarm_state":0,
"temperature":25.12,
"supervision_reports":true,
"restore_reports":true,
"test":false
}
```

Added this to my devices.js file

```

    {
       zigbeeModel: ['SMSZB-120'],
       model: 'SMSZB-120',
       vendor: 'Develco Products A/S',
       description: 'Smoke detector with siren',
       supports: 'smoke, warning, temperature',
       fromZigbee: [fz.temperature, fz.bitron_battery_att_report, fz.develco_smoke, fz.ignore_basic_report, fz.ignore_genOta],
       toZigbee: [tz.warning],
       meta: {configureKey: 1},
       configure: async (device, coordinatorEndpoint) => {
	   const endpoint = device.getEndpoint(35);
	   await bind(endpoint, coordinatorEndpoint, ['genPowerCfg', 'ssIasZone', 'genBasic']);
	   await configureReporting.batteryVoltage(endpoint);
           const endpoint2 = device.getEndpoint(38);
           await bind(endpoint2, coordinatorEndpoint, ['msTemperatureMeasurement']);
  	   await configureReporting.temperature(endpoint2);
       },
    },


```
And added this to my fromZigbee.js file

```
    develco_smoke: {
        cluster: 'ssIasZone',
        type: ['attributeReport', 'readResponse', 'commandStatusChangeNotification'],
        convert: (model, msg, publish, options, meta) => {
            const zoneState = msg.data.zoneState;
            const zoneType = msg.data.zoneType;
            const zoneStatus = msg.data.zonestatus;
            const iasCieAddr = msg.data.iasCieAddr;
            const zoneId = msg.data.zoneId;
        
        return {
            sensor_type: zoneType,
            enrolled: zoneState,
            smoke: (zoneStatus & 1) > 0, // Bit 0 = Alarm: Smoke
            IASCIEAddress: iasCieAddr,
            zoneID: zoneId,
            battery_low: (zoneStatus & 1 << 3) > 0, // Bit 3 = Battery LOW indicator
	    supervision_reports: (zoneStatus & 1 << 4) > 0,
	    restore_reports: (zoneStatus & 1 << 5) > 0,
            test: (zoneStatus & 1 << 8) > 0,

            };    
        },
    },

```

**Comment by Koenkk** [comment]
- How did you got the device working now?
- Instead of `develco_smoke` I would recommend to use the already existing converter `ias_smoke_alarm_1`
- For the siren try to add the following `toZigbee` converter: `tz.warning` and now send to `zigbee2mqtt/NEO Sirene/set` with payload `{"warning": {"mode": "emergency"}}`

**Comment by BlizzWave** [comment]
> * How did you got the device working now?

I thought that a device needed a successful interview to work. Therefor I thought that I could not troubleshoot.

> * Instead of `develco_smoke` I would recommend to use the already existing converter `ias_smoke_alarm_1`

They are not the same. 
The Develco Smoke alarm does not have tamper, trouble or ac_status.
Ref. Tech Manual

> * For the siren try to add the following `toZigbee` converter: `tz.warning` and now send to `zigbee2mqtt/NEO Sirene/set` with payload `{"warning": {"mode": "emergency"}}`

Nothing happens. 

```
zigbee2mqtt:debug 2020-03-02 19:31:21: Received MQTT message on 'zigbee2mqtt2/0x0015bc0031004688/set' with data '{"warning": {"mode": "emergency"}}'
zigbee2mqtt:debug 2020-03-02 19:31:21: Publishing 'set' 'warning' to '0x0015bc0031004688'
zigbee2mqtt:error 2020-03-02 19:31:36: Publish 'set' 'warning' to '0x0015bc0031004688' failed: 'Error: Command 0x0015bc0031004688/1 ssIasWd.startWarning({"startwarninginfo":53,"warningduration":10}, {"timeout":6000,"manufacturerCode":null,"disableDefaultResponse":false}) failed (Error: Timeout - 51981 - 1 - 2 - 1282 - 11 after 6000ms)'
zigbee2mqtt:debug 2020-03-02 19:31:36: Error: Command 0x0015bc0031004688/1 ssIasWd.startWarning({"startwarninginfo":53,"warningduration":10}, {"timeout":6000,"manufacturerCode":null,"disableDefaultResponse":false}) failed (Error: Timeout - 51981 - 1 - 2 - 1282 - 11 after 6000ms)
    at Endpoint.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/model/endpoint.js:344:23)
    at Generator.throw (<anonymous>)
    at rejected (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/model/endpoint.js:6:65)
zigbee2mqtt:info  2020-03-02 19:31:36: MQTT publish: topic 'zigbee2mqtt2/bridge/log', payload '{"type":"zigbee_publish_error","message":"Publish 'set' 'warning' to '0x0015bc0031004688' failed: 'Error: Command 0x0015bc0031004688/1 ssIasWd.startWarning({\"startwarninginfo\":53,\"warningduration\":10}, {\"timeout\":6000,\"manufacturerCode\":null,\"disableDefaultResponse\":false}) failed (Error: Timeout - 51981 - 1 - 2 - 1282 - 11 after 6000ms)'","meta":{"friendly_name":"0x0015bc0031004688"}}'
zigbee2mqtt:debug 2020-03-02 19:34:25: Received Zigbee message from '0x0015bc0031004688', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"zonestatus":48,"extendedstatus":0}' from endpoint 35 with groupID 0
```

**Comment by Koenkk** [comment]
Can you provide the `database.db` entry of this device? And are you sure it's possible to manually activate the alarm?

**Comment by BlizzWave** [comment]
Here you go:

`{"id":2,"type":"EndDevice","ieeeAddr":"0x0015bc0031004688","nwkAddr":51981,"manufId":4117,"manufName":"Develco Products A/S","powerSource":"Battery","modelId":"SMSZB-120","epList":[1,35,38],"endpoints":{"1":{"profId":49353,"epId":1,"devId":1,"inClusterList":[3,5,6],"outClusterList":[],"clusters":{},"binds":[]},"35":{"profId":260,"epId":35,"devId":1026,"inClusterList":[0,1,3,15,32,1280,1282],"outClusterList":[10,25],"clusters":{"genBasic":{"attributes":{"modelId":"SMSZB-120","manufacturerName":"Develco Products A/S","powerSource":3,"zclVersion":1,"dateCode":"2018-11-26 06:22"}},"ssIasZone":{"attributes":{"iasCieAddr":"0x00124b00193e5093","zoneState":0}},"genPowerCfg":{"attributes":{"batteryVoltage":30,"batteryAlarmState":0}}},"binds":[]},"38":{"profId":260,"epId":38,"devId":770,"inClusterList":[0,3,1026],"outClusterList":[3],"clusters":{"msTemperatureMeasurement":{"attributes":{"measuredValue":2287}}},"binds":[]}},"dateCode":"2018-11-26 06:22","zclVersion":1,"interviewCompleted":false,"meta":{"configured":1},"lastSeen":1583231344532}`

According to the technical manual you should be able to trigger the alarm. 
Currently I'm testing if they trigger each other when there is smoke.  ( I have multiple Develco smoke alarms)

**Comment by Koenkk** [comment]
The `tz.warning` uses the IAS WD cluster. Maybe the value is not correct, please try with different combinations:

So to `zigbee2mqtt/NEO Sirene/set` with payload `{"warning": {"mode": "MODE", "level": "LEVEL"}}`

- Where`MODE` can be: stop, burglar, fire, emergency, police_panic, fire_panic, emergency_panic
- And `LEVEL` can be: low, medium, high, very_high

**Comment by BlizzWave** [comment]
I tried every combination you can think of now, still no go.
On thing though, I cannot get the enroll to work, do you think that hos something to do with it?

**Comment by Koenkk** [comment]
@BlizzWave I'm not sure if it has something to do with it. Are you sure triggering the alarm actually works via Deconz?

**Comment by Koenkk** [comment]
Can you try with `zigbee2mqtt/NEO Sirene/set` with payload `{"warning": {"mode": "burglar", "level": "high", "strobe": false, "duration": 300}}`

**Comment by BlizzWave** [comment]
Still nothing

```
Mar 06 08:01:32 raspberrypi npm[9343]: zigbee2mqtt:debug 2020-03-06 08:01:32: Received MQTT message on 'zigbee2mqtt2/0x0015bc0031004688/set' with data '{"warning": {"mode": "burglar", "level": "high", "strobe": false, "duration": 300}}'
Mar 06 08:01:32 raspberrypi npm[9343]: zigbee2mqtt:debug 2020-03-06 08:01:32: Publishing 'set' 'warning' to '0x0015bc0031004688'
Mar 06 08:01:47 raspberrypi npm[9343]: zigbee2mqtt:error 2020-03-06 08:01:47: Publish 'set' 'warning' to '0x0015bc0031004688' failed: 'Error: Command 0x0015bc0031004688/1 ssIasWd.startWarning({"startwarninginfo":18,"warningduration":300}, {"timeout":6000,"manufacturerCode":null,"disableDefaultResponse":false}) failed (Error: Timeout - 14844 - 1 - 25 - 1282 - 11 after 6000ms)'
Mar 06 08:01:47 raspberrypi npm[9343]: zigbee2mqtt:debug 2020-03-06 08:01:47: Error: Command 0x0015bc0031004688/1 ssIasWd.startWarning({"startwarninginfo":18,"warningduration":300}, {"timeout":6000,"manufacturerCode":null,"disableDefaultResponse":false}) failed (Error: Timeout - 14844 - 1 - 25 - 1282 - 11 after 6000ms)
Mar 06 08:01:47 raspberrypi npm[9343]:     at Endpoint.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/model/endpoint.js:344:23)
Mar 06 08:01:47 raspberrypi npm[9343]:     at Generator.throw (<anonymous>)
Mar 06 08:01:47 raspberrypi npm[9343]:     at rejected (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/model/endpoint.js:6:65)
Mar 06 08:01:47 raspberrypi npm[9343]: zigbee2mqtt:info  2020-03-06 08:01:47: MQTT publish: topic 'zigbee2mqtt2/bridge/log', payload '{"type":"zigbee_publish_error","message":"Publish 'set' 'warning' to '0x0015bc0031004688' failed: 'Error: Command 0x0015bc0031004688/1 ssIasWd.startWarning({\"startwarninginfo\":18,\"warningduration\":300}, {\"timeout\":6000,\"manufacturerCode\":null,\"disableDefaultResponse\":false}) failed (Error: Timeout - 14844 - 1 - 25 - 1282 - 11 after 6000ms)'","meta":{"friendly_name":"0x0015bc0031004688"}}'
```

**Comment by SwoopX** [comment]
Would it help if I provide you guys with a traffic sniff while triggering the alert?

**Comment by BlizzWave** [comment]
> Would it help if I provide you guys with a traffic sniff while triggering the alert?

If you wouldn't mind that would be very helpful indeed!

**Comment by SwoopX** [comment]
Can do that in the evening.

**Comment by BlizzWave** [comment]
So i decided to make CC2531 sniffer because I was wondering what was going in with the enrollment not working. 
Some funny stuff here also. 
Error from Wireshark:

`ZCL IAS Zone: Zone Status Change Notification, Seq: 4[Malformed Packet]`

**Comment by SwoopX** [comment]
Here you go

![grafik](https://user-images.githubusercontent.com/4005212/76117808-cfb6b680-5fec-11ea-88d5-a1fc3215f8bf.png)

**Comment by BlizzWave** [comment]
Thank you!

Here is how it looks when zigbee2mqtt is sending the command.
But these seem different
`Acknowlegment Request`
`Disable Default Resonse `

![sniff](https://user-images.githubusercontent.com/6298655/76118263-c8dc7380-5fed-11ea-9150-46c4e4b9f986.JPG)

**Comment by SwoopX** [comment]
Hm, if I recall correctly, the default response thingy could cause hickups... Also, I had real difficulties to get the device enrolled back in the days. Not sure if that has been sorted out by now.

**Comment by BlizzWave** [comment]
I have just modified line 384 in endpoint.js so that I would get a "true" on the Disable Default Response.

`return Object.assign({ timeout: 6000, manufacturerCode: null, disableDefaultResponse: true }, providedOptions);`

This did not help.

Also the `Frame Control Field: Cluster -specific` is different.

I'm hoping @Koenkk will figure something out.

**Comment by Koenkk** [comment]
@SwoopX sniffer log shows that the message is send to endpoint 35, while yours is send to 1. 

Try with:

```js
{
    zigbeeModel: ['SMSZB-120'],
    model: 'SMSZB-120',
    vendor: 'Develco Products A/S',
    description: 'Smoke detector with siren',
    supports: 'smoke, warning, temperature',
    fromZigbee: [fz.temperature, fz.bitron_battery_att_report, fz.develco_smoke, fz.ignore_basic_report, fz.ignore_genOta],
    toZigbee: [tz.warning],
    meta: {configureKey: 1},
    configure: async (device, coordinatorEndpoint) => {
        const endpoint = device.getEndpoint(35);
        await bind(endpoint, coordinatorEndpoint, ['genPowerCfg', 'ssIasZone', 'genBasic']);
        await configureReporting.batteryVoltage(endpoint);
        const endpoint2 = device.getEndpoint(38);
        await bind(endpoint2, coordinatorEndpoint, ['msTemperatureMeasurement']);
        await configureReporting.temperature(endpoint2);
    },
    endpoint: (device) => {
        return {'default': 35};
    },
    },
```

Also, please remove the `disableDefaultResponse` that you've added.

**Comment by SwoopX** [comment]
@Koenkk good catch. Haven't thought about it since I don't need to worry with deconz. Of course it must go to 35, 1 is the manufacturer specific endpoint.

**Comment by BlizzWave** [comment]
Well I must be blind or something. Did not see that.

**BUT**
It works now!

Thanks for the help @Koenkk and @SwoopX 

To get the siren to start
`{"warning": {"mode": "burglar", "level": "high", "strobe": false, "duration": 300}}`

To stop it
`{"warning": {"mode": "stop", "level": "low", "strobe": false, "duration": 300}}`

It gives a error when starting the siren, but it works.

```
Mar 08 19:32:44 raspberrypi npm[856]: zigbee2mqtt:error 2020-03-08 19:32:44: Publish 'set' 'warning' to '0x0015bc0031004688' failed: 'Error: Command 0x0015bc0031004688/35 ssIasWd.startWarning({"startwarninginfo":18,"warningduration":3}, {"timeout":6000,"manufacturerCode":null,"disableDefaultResponse":false}) failed (Error: AREQ - AF - dataConfirm after 3000ms)'
Mar 08 19:32:44 raspberrypi npm[856]: zigbee2mqtt:debug 2020-03-08 19:32:44: Error: Command 0x0015bc0031004688/35 ssIasWd.startWarning({"startwarninginfo":18,"warningduration":3}, {"timeout":6000,"manufacturerCode":null,"disableDefaultResponse":false}) failed (Error: AREQ - AF - dataConfirm after 3000ms)
Mar 08 19:32:44 raspberrypi npm[856]:     at Endpoint.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/model/endpoint.js:344:23)
Mar 08 19:32:44 raspberrypi npm[856]:     at Generator.throw (<anonymous>)
Mar 08 19:32:44 raspberrypi npm[856]:     at rejected (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/model/endpoint.js:6:65)
Mar 08 19:32:44 raspberrypi npm[856]: zigbee2mqtt:info  2020-03-08 19:32:44: MQTT publish: topic 'zigbee2mqtt2/bridge/log', payload '{"type":"zigbee_publish_error","message":"Publish 'set' 'warning' to '0x0015bc0031004688' failed: 'Error: Command 0x0015bc0031004688/35 ssIasWd.startWarning({\"startwarninginfo\":18,\"warningduration\":3}, {\"timeout\":6000,\"manufacturerCode\":null,\"disableDefaultResponse\":false}) failed (Error: AREQ - AF - dataConfirm after 3000ms)'","meta":{"friendly_name":"0x0015bc0031004688"}}'

```

**Comment by Koenkk** [comment]
Can you modify the toZigbee warning converter to add a timeout like done here: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/converters/toZigbee.js#L844 ? (check if it works with a timeout of `10000`).

**Comment by SwoopX** [comment]
One additional thing though, as I'm pretty sure it does not work with deconz. Not sure if it has been taken into consideration here. It's about clearing the alarm state after the smoke alarm has been triggered.

![grafik](https://user-images.githubusercontent.com/4005212/76170194-dde90c00-617f-11ea-8d87-a37fbf22967f.png)

**Comment by BlizzWave** [comment]
> Can you modify the toZigbee warning converter to add a timeout like done here: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/converters/toZigbee.js#L844 ? (check if it works with a timeout of `10000`).

Like this?

```
const info = (mode[values.mode] << 4) + ((values.strobe ? 1 : 0) << 2) + (level[values.level]);
const opts = {timeout: 10000}; //Test


await entity.command(
    'ssIasWd',
    'startWarning',
    {startwarninginfo: info, warningduration: values.duration, opts},
    getOptions(meta),
```

**Comment by Koenkk** [comment]
Great, once you feel it's finished, could you make the PRs?

**Comment by BlizzWave** [comment]
Yes, I will do that.
They way I modified the warning config effects all devices that use tz.warning. 
Should I create a separate config for the Develco ex. tz.warning_develco or 
should we have exception inside the current tz.warning?
The latter I don't know how to do. I'm not good at programming! 

Next thing is to look at is the enrollment issue. 
For a fire alarm to be really good it needs to be able to start other fire alarms without the need to go to the coordinator first. 
And if I understand the technical manual it should be able to do that.

@SwoopX Does enrollment work on Deconz?

**Comment by Koenkk** [comment]
Just create the PRs and I will do the cleanups.

**Comment by SwoopX** [comment]
@BlizzWave yes, it's working flawlessly. But as I said, I had trouble with zigbee2mqtt back in the days. Could that also be related to the endpoint? IAS stuff is endpoint 35...

**Comment by Koenkk** [comment]
@BlizzWave to check please sniff the traffic when pairing the device with zigbee2mqtt.

**Comment by BlizzWave** [comment]
Here's the log from Wireshark.

[develco_fire_alarm_sniffing.zip](https://github.com/Koenkk/zigbee2mqtt/files/4314773/develco_fire_alarm_sniffing.zip)

**Comment by Koenkk** [comment]
I'm not sure this was just an incident, but the status change notification, where zigbee-herdsman is waiting on to confirm is has been enrolled correctly, is malformed:

![image](https://user-images.githubusercontent.com/2892853/76445598-f480a480-63c5-11ea-8d92-84e9814e3e8d.png)

Could you sniff the traffic one more and confirm that the package is malformed again? If yes, I guess that the device is enrolled just fine (as it also triggers on smoke).

**Comment by BlizzWave** [comment]
Its not random. I get that every time.
The smoke alarm does not set the enroll to true either.
I have not tested with smoke to see if it triggers the other fire alarms yet though.

I quote myself from ealier:
> Some funny stuff here also.
> Error from Wireshark:
> 
> `ZCL IAS Zone: Zone Status Change Notification, Seq: 4[Malformed Packet]`

**Comment by Koenkk** [comment]
But the alarm is triggered when there is smoke, right?

Sorry missed the malformed packet part, but in that case it's expected that interview fails, if the device works as expected this can be ignored though (in that case add it to the docs).

**Comment by BlizzWave** [comment]
Yes, it triggers.  (Will do test with multiple alarms to check if they trigger each other later)
The interview fails sometimes and you can try again and it works. 
I can remember reading something about this in another post, that the smoke alarm needs a couple of tries to actually interview correctly.

**Comment by Koenkk** [comment]
Good, as we cannot the interview fail issue (device issue due to the malformed packet), could you make a PR?

**Comment by BlizzWave** [comment]
Well I'm wondering if this happens on Deconz also, if its the device then Deconz should have the same error

@SwoopX could I bother you once more to do sniff traffic again while the sensor is being interviewed?

**Comment by SwoopX** [comment]
What do you guys need exactly? Enrollment is working well on deconz side

![grafik](https://user-images.githubusercontent.com/4005212/76455909-8a243000-63d6-11ea-8bac-c45f6f66cf59.png)

However, I also see malformed packets, but typically directly after a valid one. Afaik, the malformed has an increased counter by 2.

![grafik](https://user-images.githubusercontent.com/4005212/76456502-5f86a700-63d7-11ea-8477-c597427d747b.png)

**Comment by BlizzWave** [comment]
Could post the whole log like I did?

**Comment by SwoopX** [comment]
[Develco_join_neu3.zip](https://github.com/Koenkk/zigbee2mqtt/files/4320553/Develco_join_neu3.zip)

**Comment by BlizzWave** [comment]
@SwoopX That can't be correct. Contains no real Zigbee traffic

**Comment by BlizzWave** [comment]
I also get the Enroll request and Enroll response. But it keeps on asking for enroll even though it has gotten a successful response. 
And i get a malformed packet after a while, same as @SwoopX .
I have also tested with smoke to see if they trigger each other. They dont **NOT**.

@SwoopX your enroll works, but if one alarm triggers from smoke does it start the other alarms?
I'm assuming you have multiple. haha

**Comment by SwoopX** [comment]
> That can't be correct. Contains no real Zigbee traffic

It is, Dedicated test gateway with only the sensor in it.

> your enroll works, but if one alarm triggers from smoke does it start the other alarms?

Haven't tested that one, I currently don't dare it. Got 4 of them operational, two more lying around.

**Comment by BlizzWave** [comment]
> > That can't be correct. Contains no real Zigbee traffic
> 
> It is, Dedicated test gateway with only the sensor in it.
Okay, I can't see it. If I sort by protocol I only get beacons from the Deconz. 
No joining or reporting of any sensor.

Might be asking alot, but the ones you have laying around, could you join them to your test network and see if they trigger each other from smoke?

Because if they dont then there is a issue with the firmware in the smoke alarm and I will send a email to Develco Support.

**Comment by SwoopX** [comment]
I knew this request would come... :P

I'll see what I can do on the weekend. Regarding the firmware, I know there are, well, "bugs" or inconsistencies. Was also thinking of reaching out trying to get access to their support forum. Not sure if some observations are known and eventually resolved by any updates. Maybe we can team up then...

**Comment by BlizzWave** [comment]
> I knew this request would come... :P
> 
> I'll see what I can do on the weekend. Regarding the firmware, I know there are, well, "bugs" or inconsistencies. Was also thinking of reaching out trying to get access to their support forum. Not sure if some observations are known and eventually resolved by any updates. Maybe we can team up then...

Haha! I had to ask. Thanks for your help so far!

Btw I have contacted Develco. Hopefully I get some response. I'll l post what I get here. 
If (really big if here) theres some kind of firmware update for this issue. Then we stuble on a new issue, how would we update the smoke alarm?

**Comment by BlizzWave** [comment]
@SwoopX Did you get to test it?

A little update, I have been in contact with Develco about the enrollment issue. They are looking into it.
Btw, to get access to their support forum you need to pay a fee.

**Comment by SwoopX** [comment]
> A little update, I have been in contact with Develco about the enrollment issue. They are looking into it.
> Btw, to get access to their support forum you need to pay a fee.

Man, I just wanna have the firmware updates and the "bugs" fixed they have...

Haven't had the time to check it unfortunately. The meltdown out there made life a bit more complicated...

**Comment by BlizzWave** [comment]
Got a response from Develco today. 
@Koenkk what do you think? 

> Issue is that your Gateway is taking to long to answer the Enrollrequest.
> 
> Our sensors gives the TIMEOUT 0x94 (The exchange was aborted due to excessive response time.) after 15 seconds.
> 
>  
> 
> ------------------------------------
> Lasse Pedersen
> Senior Technical Adviser
> Phone: +45 30 54 30 46
> Develco Products A/S

**Comment by Koenkk** [comment]
I investigated your sniff a bit more and think we can one more thing.

- First update to zigbee2mqtt 1.12.0
- Now replace `/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/controller.js` with the following https://gist.github.com/Koenkk/6cc1d83b631775ff67c5da9039c9ca33
- Pair the device again (factory reset it) and provide both the sniff log and zigbee herdsman debug log

**Comment by BlizzWave** [comment]
Followed your instructions. 
The sensor will now enroll, but only for a couple of seconds.
Also the timeout 0x94 is still there.

But have a look in the logs

Attached sniff and debug. 
[sniff_and_debug.zip](https://github.com/Koenkk/zigbee2mqtt/files/4362030/sniff_and_debug.zip)

**Comment by SwoopX** [comment]
@BlizzWave check out https://github.com/dresden-elektronik/deconz-rest-plugin/issues/2608#issuecomment-601749756 regarding the group alert.

**Comment by Koenkk** [comment]
In addition to the controller.js from https://github.com/Koenkk/zigbee2mqtt/issues/3012#issuecomment-601871251 can you also update the `/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/model/device.js` with https://gist.github.com/Koenkk/6999e024118f454df7303a453c7017a8

Interview should succeed after this.

Btw: where do you see the 0x94 timeout error in that sniff?

**Comment by BlizzWave** [comment]
> @BlizzWave check out [dresden-elektronik/deconz-rest-plugin#2608 (comment)](https://github.com/dresden-elektronik/deconz-rest-plugin/issues/2608#issuecomment-601749756) regarding the group alert.
> 
Will have a look. Thanks

> Btw, have you sent your request to [info@develcoproducts.com](mailto:info@develcoproducts.com)? Only regarding the enrollment issue or also regarding the malformed packet?

Yes, I have. We have so far talked about the enrollment issue. But I will start asking more about the malformed packet now since this must be a firmware issue on their side.
 
> Would like to engage them for a node descriptor issue as well (and hopefully get a firmware update ;)).

@Koenkk 

I think you got the enroll to work now. Bit for enroll is set to 1.
I thought that triggering one of the sensors should start the other one when properly enrolled.
But the second sensor does not react. 
Will do some sniffing to see what actually happens. 

> In addition to the controller.js from [#3012 (comment)](https://github.com/Koenkk/zigbee2mqtt/issues/3012#issuecomment-601871251) can you also update the `/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/model/device.js` with https://gist.github.com/Koenkk/6999e024118f454df7303a453c7017a8
> 
> Interview should succeed after this.
> 
> Btw: where do you see the 0x94 timeout error in that sniff?

Look at no 157 in the Wireshark sniff.

**Comment by BlizzWave** [comment]
Sensor being triggered by smoke.
Reports smoke but also the famous malformed packet comes in.
I just sent a email to Develco about this, but this must be firmware related.

![triggered by smoke](https://user-images.githubusercontent.com/6298655/77237914-6f1ea080-6bcc-11ea-8ce0-f20b51ce5b7b.JPG)

Malformed Packet

![triggered by smoke (malformed)](https://user-images.githubusercontent.com/6298655/77237915-704fcd80-6bcc-11ea-99dd-52ac89ac95e2.JPG)

**Comment by Koenkk** [comment]
Good, these changes are available in the latest dev branch now so the device should pair fine without any changes. Could you make a PR to zigbee-herdsman-converters to add support for this device?

Regarding auto triggering more smoke alarms: I didn't see this yet within Zigbee, atm it makes sense that it doesn't work as the smoke sensor only sends messages to the coordinator.

**Comment by BlizzWave** [comment]
So to get this to work is to make a "automation" in whatever home automation you might use and manually trigger the other sensors with the command from earlier in the thread.

> Regarding auto triggering more smoke alarms: I didn't see this yet within Zigbee, atm it makes sense that it doesn't work as the smoke sensor only sends messages to the coordinator.

**Comment by BlizzWave** [comment]
Response from Develco stopped so I just did the PR's for this. If theres any response in the future I will post here and update whats needed

**Comment by SwoopX** [comment]
Maybe I just kept them too busy with my request on the node descriptor 😜

**Comment by BlizzWave** [comment]
So I have finally integrated the smoke alarms from Develco in my Zigbee2MQTT / Home Assistant system.

There one small annoyance though.
The sensor does report voltage, but it seems Zigbee2MQTT does not convert it to battery percentage when reporting to Home Assistant.

**Comment by Koenkk** [comment]
@BlizzWave is this with the latest dev branch? (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html) I see the device has a flag set to convert the voltage to percentage (https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices.js#L11184). 

Note that when switching you need to wait for a new battery voltage report before z2m calculates the %.

**Comment by BlizzWave** [comment]
No, its the stable release.

This is what it reports to Zigbee2MQTT
````
{"voltage":3100,"battery_low":false,"linkquality":138,"enrolled":false,"smoke":false,"tamper":false,"supervision_reports":true,"restore_reports":true,"trouble":false,"ac_status":false,"temperature":22.81}
````
In devices.js is says
````
await configureReporting.batteryVoltage(endpoint);

````
Shouldn't it be like this ?
````
await configureReporting.batteryPercentageRemaining(endpoint);
````

**Comment by Koenkk** [comment]
@BlizzWave you can try if that works, after making that change manually reconfigure through https://www.zigbee2mqtt.io/information/mqtt_topics_and_message_structure.html#zigbee2mqttbridgeconfigure check the log if configure is succesfull.

**Comment by BlizzWave** [comment]
Tried it, but gives this error. 

````
Aug 06 11:47:44 zigbee-coordinator npm[15493]: zigbee2mqtt:error 2020-08-06 11:47:44: Failed to configure '0x0015bc003100568d', attempt 1 (Error: ConfigureReporting 0x0015bc003100568d/35 genPowerCfg([{"attribute":"batteryPercentageRemaining","minimumReportInterval":3600,"maximumReportInterval":62000,"reportableChange":0}], {"timeout":10000,"disableResponse":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null}) failed (Status 'UNSUPPORTED_ATTRIBUTE')

````

**Comment by Koenkk** [comment]
It's indeed not supported (that's why we use voltage). Anyway the conversion of voltage to battery has been implemtend in https://github.com/Koenkk/zigbee-herdsman-converters/pull/1426/files (which is not in 1.14.2, only in dev branch for now)

**Comment by BlizzWave** [comment]
Perfect!

**Comment by TheStigh** [comment]
@BlizzWave Sorry for waking this issue up, but did you ever resolve the enrollment issue ?

**Comment by TheStigh** [comment]
@BlizzWave @SwoopX 

I've amanged to get the latest fw (4.0.2) from Develco and added it to Z2M OTA.
I've updated the device to support OTA (and did remove ac_status, tamper and trouble).
Make sure you get the SMSZB-120 re-configured, now you should also see current fw as an attribute.

I'm not good with the sniffer so I do hope you guys would pick up this thread and have a look if you get the enrollment going?

**Comment by SwoopX** [comment]
@TheStigh What on earth have you told them they gave you the firmware? I need to get my hands on that as well.

Have you already tried writing the coordinators MAC address to the IAS CIE attribute?

**Comment by TheStigh** [comment]
> 
> 
> @TheStigh What on earth have you told them they gave you the firmware? I need to get my hands on that as well.
> 
> Have you already tried writing the coordinators MAC address to the IAS CIE attribute?

@SwoopX 

You should see my emails - but I did also speak with them :)
To get the update, just re-configure and go to OTA in UI and do the update. But it takes approx 4 hrs!

No, I haven't tried - honestly I am not to familiar with IAS Zone-stuff...

**Comment by TheStigh** [comment]
@SwoopX @BlizzWave @Koenkk 

Interesting, after FW update from 3.4.x to 4.0.2, I now get as shown below during pairing - both on the 'old' and my 'new' device. Seems they have done some changes on `iasCie `which again can give a hope for supported enrollment... Just need to find some time to look into this.

```
Zigbee2MQTT:debug 04-07-2021 17:48:39: Received Zigbee message from '0x0015bc0031005194', type 'readResponse', cluster 'ssIasZone', data '{"iasCieAddr":"0x00124b0018e26fb0","zoneState":1}' from endpoint 35 with groupID 0
Zigbee2MQTT:debug 04-07-2021 17:48:39: No converter available for 'SMSZB-120' with cluster 'ssIasZone' and type 'readResponse' and data '{"iasCieAddr":"0x00124b0018e26fb0","zoneState":1}'
```

**Comment by SwoopX** [comment]
You might wanna ping me on discord (https://discord.gg/na4ccVnY) for some more interactive exchange. I maybe can help you things get going and explain a bit.

**Comment by heinekmadsen** [comment]
@TheStigh 

> @SwoopX @BlizzWave @Koenkk
> 
> Interesting, after FW update from 3.4.x to 4.0.2, I now get as shown below during pairing - both on the 'old' and my 'new' device. Seems they have done some changes on `iasCie `which again can give a hope for supported enrollment... Just need to find some time to look into this.
> 
> ```
> Zigbee2MQTT:debug 04-07-2021 17:48:39: Received Zigbee message from '0x0015bc0031005194', type 'readResponse', cluster 'ssIasZone', data '{"iasCieAddr":"0x00124b0018e26fb0","zoneState":1}' from endpoint 35 with groupID 0
> Zigbee2MQTT:debug 04-07-2021 17:48:39: No converter available for 'SMSZB-120' with cluster 'ssIasZone' and type 'readResponse' and data '{"iasCieAddr":"0x00124b0018e26fb0","zoneState":1}'
> ```

Did you happen to find some time to look into the iasCie? I assume it's for pairing the alarms togeater and then they can activate each other, or am i wrong here?

**Comment by jkaberg** [comment]
Also interested in this getting "grouping" without coordinator going (if possible). Referring to the technical manual (over [here](https://www.manualslib.com/manual/1236036/Develco-Smszb-120.html?page=9#manual) - see chapter 4.2 and especially 4.2.4.2 and 4.2.4.2.1) there seems to be functionality to group these devices and trigger the siren without the coordinator? 

Anyone with more insight care to shed light on this and what the current status is?

**Comment by bouyssic** [comment]
Hello team,

Just got 2 of those for home and I'm experiencing the same with the first I tried linking:

`Error Failed to configure 'Living Room Fire Alarm', attempt 4 (Error: Bind 0x0015bc003100ad92/35 genBasic from '0x00212effff0661ad/1' failed (Error: status: 140) at DeconzAdapter.bind (/app/node_modules/zigbee-herdsman/src/adapter/deconz/adapter/deconzAdapter.ts:753:19) at Endpoint.bind (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:517:13) at Object.bind (/app/node_modules/zigbee-herdsman-converters/lib/reporting.js:33:9) at Object.configure (/app/node_modules/zigbee-herdsman-converters/devices/develco.js:165:13) at Configure.configure (/app/lib/extension/configure.ts:115:13) at Configure.onMQTTMessage (/app/lib/extension/configure.ts:55:21))
`
As per other users I'm using a conbeeII with deconz adapter. I'll keep digging see if I can find something else. 

Has anybody solved this?
Thanks

**Comment by bouyssic** [comment]
I've just tried with Edge and it's still not working

I guess it's something else than Z2M causing this issue

**Comment by bouyssic** [comment]
I've just noticed my device is a Frient, not a Develco. I don't know if it is a rebranding or a different device. I might need to walk the "Add support for a device route"

**Comment by TheStigh** [comment]
> I've just noticed my device is a Frient, not a Develco. I don't know if it is a rebranding or a different device. I might need to walk the "Add support for a device route"

Integration are made with Z2M recommended coordinators. Conbee are just experimental.

**Comment by bouyssic** [comment]
OOOkay. What else should I do in your opinion then?

**Comment by bouyssic** [comment]
Sorry @TheStigh ... posted on the wrong thread.

**Comment by nrosier** [comment]
Got some new devices and these come with firmware 4.0.4. What's the procedure to update the firmware in the repo to this newer one?

**Comment by diggidiboy** [comment]
I'm having the exact same issues, but I don't have enough knowledge to follow your steps

**Comment by TheStigh** [comment]
What issues? You should tell exactly your issues when posting for help. This thread are old, and long.

**Comment by boelle** [comment]
> 11\. "temperature" is missing (null), "reliability" is missing (null), "fault" is missing (null)

i have this one too.... on one of 2  smoke detectors (i got 2 of the same one)
